component/src/main/java/com/celements/photo/utilities/DefaultImageUrlExtractor.java

```diff
@@ -0,0 +1,193 @@
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+@Immutable
+@Singleton
+@Component
+public class DefaultImageUrlExtractor implements ImageUrlExtractor {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(DefaultImageUrlExtractor.class);
+
+  public static final int MAX_ALLOWED_DIM = 1000000000;
+  public static final int MIN_SOCIAL_MEDIA_IMAGE_SIZE = 200;
+  public static final long MIN_SOCIAL_MEDIA_AREA_SIZE = MIN_SOCIAL_MEDIA_IMAGE_SIZE
+      * MIN_SOCIAL_MEDIA_IMAGE_SIZE;
+
+  private final Pattern IMG_FROM_HTML_PATTERN = Pattern.compile("<img .*?src=['\"](.*?)['\"].*?/>");
+
+  @Override
+  public @NotNull List<ImageUrl> extractImageUrlList(@NotNull String content) {
+    Preconditions.checkNotNull(content);
+    List<ImageUrl> imageUrls = new ArrayList<>();
+    Matcher m = IMG_FROM_HTML_PATTERN.matcher(content);
+    while (m.find()) {
+      try {
+        imageUrls.add(new ImageUrl.Builder().url(m.group(1)).build());
+      } catch (IllegalImageUrlException iiue) {
+        LOGGER.info("ImageUrl only works for relative URLs. Failed for {}", iiue.getUrl());
+      }
+    }
+    return imageUrls;
+  }
+
+  @Override
+  public @NotNull List<ImageUrl> extractImagesSocialMediaUrlList(@NotNull String content) {
+    List<ImageUrl> imageUrls = extractImageUrlList(content);
+    return getImagesListForSizeMap(filterMinMaxSize(groupImageUrlsBySize(imageUrls), Optional.of(
+        MIN_SOCIAL_MEDIA_IMAGE_SIZE), Optional.<Integer>absent(), Optional.of(
+            MIN_SOCIAL_MEDIA_AREA_SIZE), Optional.<Long>absent(), Optional.of(-1L)), true, false);
+  }
+
+  @Override
+  public @NotNull Map<Long, List<ImageUrl>> groupImageUrlsBySize(
+      @NotNull List<ImageUrl> imageUrls) {
+    Map<Long, List<ImageUrl>> articleImages = new TreeMap<>();
+    for (ImageUrl imgUrl : imageUrls) {
+      Long key = getImgUrlSizeKey(imgUrl);
+      if (!articleImages.containsKey(key)) {
+        articleImages.put(key, new ArrayList<ImageUrl>());
+      }
+      articleImages.get(key).add(imgUrl);
+    }
+    return articleImages;
+  }
+
+  @Override
+  public @NotNull Map<Long, List<ImageUrl>> filterMinMaxSize(
+      @NotNull Map<Long, List<ImageUrl>> groupedImageUrls,
+      @Nullable Optional<Integer> minSideLength, @Nullable Optional<Integer> maxSideLength,
+      @Nullable Optional<Long> minPixels, @Nullable Optional<Long> maxPixels,
+      @Nullable Optional<Long> keepKeyAsDefault) {
+    Map<Long, List<ImageUrl>> filteredMap = filterByPixels(groupedImageUrls, minPixels.or(1L),
+        maxPixels.or(Long.MAX_VALUE), keepKeyAsDefault);
+    filterBySideLength(filteredMap, minSideLength.or(1), maxSideLength.or(MAX_ALLOWED_DIM),
+        keepKeyAsDefault);
+    return filteredMap;
+  }
+
+  @Override
+  public @NotNull List<ImageUrl> getImagesListForSizeMap(
+      @NotNull Map<Long, List<ImageUrl>> groupedImageUrls, boolean imageSizeAsc,
+      boolean imageEncounterAsc) {
```

Comment:
rough oral answer from Fabian:
please implement, because next time we need another sorting/filtering we'll have to reconstruct it anyway, but then we'd already have to deal with deprecated methods.